### PR TITLE
API documentation and test updates

### PR DESCRIPTION
**Pull request by runforyourlives**
Add comments on each API.

Update an error message for get_scenario_by_id so it is harder to distinguish a missing plan vs. a plan that belongs to someone else.

Update tests:
  - move happy case tests up in the list for each test family.
  - add more output validation, especially for failure cases.
  - made naming of some tests more consistent across the file.
  - added more tests, fixed a few that were failing for the wrong reason.

### CODE REVIEW

File: src/planscape/planning/tests.py [hunk 1]
```diff
@@ -37,6 +37,34 @@ def setUp(self):
...
+        response = self.client.post(
+            reverse('planning:create_plan'),
+            {'name': 'plan', 'region_name': 'Southern California', 'geometry': self.multipolygon_geometry},
+            content_type='application/json')
+        self.assertEqual(response.status_code, 200)
+        plans = PlanningArea.objects.all()
+        self.assertEqual(plans.count(), 1)
+        plan = plans.first()
+        assert plan is not None
+        self.assertEqual(plan.region_name, 'southern_california')
```
Comment (sachiyuan) [inline src/planscape/planning/tests.py:67]: Should we also validate that the geometry was saved properly here?

File: src/planscape/planning/views.py [hunk 2]
```diff
@@ -306,6 +377,16 @@ def update_scenario_result(request: HttpRequest) -> HttpResponse:
 # This returns the empty set if given a plan that has no scenarios,
```
Comment (sachiyuan) [inline src/planscape/planning/views.py:377]: redundant comment

File: src/planscape/planning/views.py [hunk 3]
```diff
@@ -328,6 +409,16 @@ def list_scenarios_for_plan(request: HttpRequest) -> HttpResponse:
 # Deletion attempts for someone else's scenario will silently do nothing with those scenarios.
```
Comment (sachiyuan) [inline src/planscape/planning/views.py:409]: redundant comment

File: src/planscape/planning/views.py [hunk 4]
```diff
@@ -306,6 +377,16 @@ def update_scenario_result(request: HttpRequest) -> HttpResponse:
 # This returns the empty set if given a plan that has no scenarios,
 # a plan that isn't owned by the user, or a planID that doesn't exist.
 def list_scenarios_for_plan(request: HttpRequest) -> HttpResponse:
+    """
+    Lists all Scenarios for a Plan.
+    Requires a logged in user, as a scenario must be associated with a user's planning area.
+
+    Returns: a list of the scenarios for the user.  Can return the empty list.  Will return an empty list
+      if given a plan that has no scenarios, a plan that isn't owned by the user, or a plan ID that doesn't exist.
```
Comment (sachiyuan) [inline src/planscape/planning/views.py:385]: ', or if there is no existing plan associated with the given planning area ID' *

As another note, plan = planning area, correct? For a future edit, may be worth making the language consistent with one term or the other

File: src/planscape/planning/tests.py (hunk 1 above)
Comment (runforyourlives) [reply src/planscape/planning/tests.py:67]: yes. added.

File: src/planscape/planning/views.py (hunk 2 above)
Comment (runforyourlives) [reply src/planscape/planning/views.py:377]: removed.

File: src/planscape/planning/views.py (hunk 4 above)
Comment (runforyourlives) [reply src/planscape/planning/views.py:385]: Updated comments and switched all references from "plan" to "planning_area".

File: src/planscape/planning/views.py (hunk 3 above)
Comment (runforyourlives) [reply src/planscape/planning/views.py:409]: removed.

**Review by sachiyuan — APPROVED** [review]
lgtm
(especially approve of the test scenario results + run details)

**Pull request merged**